hw/drivers/sensors/lis2dw12/src/lis2dw12.c

```diff
@@ -1863,60 +1915,92 @@ int lis2dw12_get_int1_on_int2_map(struct sensor_itf *itf, uint8_t *val)
 int lis2dw12_run_self_test(struct sensor_itf *itf, int *result)
 {
     int rc;
-    int16_t base[3], pos[3];
+    /*configure min and max values for reading 5 samples, and accounting for
+     * both negative and positive offset */
+    int min = LIS2DW12_ST_MIN*5*2;
+    int max = LIS2DW12_ST_MAX*5*2;
+
+    int16_t data[3], diff[3] = {0,0,0};
     int i;
-    int16_t change;
-    uint8_t fs;
+    uint8_t prev_config[6];
+    /* set config per datasheet, with positive self test mode enabled. */
+    uint8_t st_config[] = {0x44, 0x08, 0x40, 0x00, 0x00, 0x10};
```

Comment:
Would be nice if this was explicit or at least documented in comments what it was trying to do. For instance youre enabling self test here, and then again later...Is that purposeful? Also you seem to set FS_4G here. Any particular reasoning? In many other implementations including the one you linked I find 2g